native/Makefile

```diff
@@ -401,7 +418,13 @@ JNISFX=.jnilib
 ifneq ($(SDKROOT),)
 SYSLIBROOT=-Wl,-syslibroot,$(SDKROOT)
 ISYSROOT=-isysroot $(SDKROOT)
-ARCHFLAGS=-arch i386 -arch x86_64
+ARCHFLAGS=-arch x86_64
+ifeq ($(shell echo "int main(void) { return 0; }" | $(CC) $(LOC_CC_OPTS) -arch i386 $(CFLAGS) -x c - -o /dev/null &>/dev/null; echo $$?),0)
+ARCHFLAGS+=-arch i386
+endif
+ifeq ($(shell echo "int main(void) { return 0; }" | $(CC) $(LOC_CC_OPTS) -arch arm64 $(CFLAGS) -x c - -o /dev/null &>/dev/null; echo $$?),0)
+ARCHFLAGS+=-arch arm64
+endif
```

Comment:
This looks as if i386 and/or arm64 are conditional - is this correct? Is my interpretation correct, that this basicly says: "If the C compiler can compile for 32bit, then do it, if it can compile for arm64, then do it. Is this correct?

Reply:
Yes, correct.

Reply:
Currently we build fatbinaries with x86 + x86-64. Is there an XCode version out there, that is able to create a tipple binary: x86 + x86_64 + arm64? Or would now be a good time to split the binaries for mac OS as all other architectures? This should be taken to the mailing list, as it will be incompatible/needs some deeper thoughts.